pkg/validate/dynamic/platformworkloadidentityprofile.go

```diff
@@ -0,0 +1,117 @@
+			}
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodePlatformWorkloadIdentityMismatch,
+				"properties.ValidatePlatformWorkloadIdentityProfile.PlatformWorkloadIdentities", "There's a mismatch between the required and expected set of platform workload identities for the requested OpenShift version '%s'. The required platform workload identities are '%v'", oc.Properties.ClusterProfile.Version, requiredOperatorIdentities)
+		}
+	}
+
+	err := dv.validateClusterMSI(ctx, oc, roleDefinitions)
+	if err != nil {
+		return err
+	}
+
+	for _, role := range platformWorkloadIdentityRoles {
+		actions, err := getActionsForRoleDefinition(ctx, role.RoleDefinitionID, roleDefinitions, http.StatusBadRequest)
+		if err != nil {
+			return err
+		}
+
+		platformIdentitiesActionsMap[role.OperatorName] = actions
+	}
+
+	dv.platformIdentities = oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities
+	dv.platformIdentitiesActionsMap = platformIdentitiesActionsMap
+
+	return nil
+}
+
+func (dv *dynamic) validateClusterMSI(ctx context.Context, oc *api.OpenShiftCluster, roleDefinitions armauthorization.RoleDefinitionsClient) error {
+	if oc.Identity == nil || len(oc.Identity.UserAssignedIdentities) != 1 {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidClusterMSICount,
+			"identity.userAssignedIdentities", "Unexpected number of OpenShift Cluster associated User Assigned Identity are provided for the Workload Identity OpenShift cluster, expected one User Assigned Identity")
+	}
+
+	for resourceID, identity := range oc.Identity.UserAssignedIdentities {
+		_, err := azure.ParseResourceID(resourceID)
+		if err != nil {
+			return err
+		}
+
+		return dv.validateClusterMSIPermissions(ctx, identity.PrincipalID, oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities, roleDefinitions)
+	}
+
+	return nil
+}
+
+func (dv *dynamic) validateClusterMSIPermissions(ctx context.Context, oid string, platformIdentities []api.PlatformWorkloadIdentity, roleDefinitions armauthorization.RoleDefinitionsClient) error {
+	actions, err := getActionsForRoleDefinition(ctx, rbac.RoleAzureRedHatOpenShiftFederatedCredentialRole, roleDefinitions, http.StatusInternalServerError)
+	if err != nil {
+		return err
+	}
+
+	for _, platformIdentity := range platformIdentities {
+		pid, err := azure.ParseResourceID(platformIdentity.ResourceID)
+		if err != nil {
+			return err
+		}
+
+		err = dv.validateActionsByOID(ctx, &pid, actions, &oid)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func getActionsForRoleDefinition(ctx context.Context, roleDefinitionID string, roleDefinitions armauthorization.RoleDefinitionsClient, errorStatusCode int) ([]string, error) {
+	definition, err := roleDefinitions.GetByID(ctx, roleDefinitionID, &sdkauthorization.RoleDefinitionsClientGetByIDOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(definition.RoleDefinition.Properties.Permissions) <= 0 {
```

Comment:
Changed the condition to `==`

Reply:
I wasn't necessarily asking for a change here. I was just curious what made you choose to use `<= 0` rather than `!= 1`, which makes the most sense to me if we're assuming that the slice should contain exactly 1 element.

Reply:
I don't have any idea why `<=` was used previously when the PR was in draft, also since there were 2 weeks of on-call in between it doesn't help to remember.
I agree that it makes no sense at all to have `<=0` and it is supposed to be `!=1`.

Reply:
> also since there were 2 weeks of on-call in between it doesn't help to remember

:laughing::laughing::laughing: